doc/advanced-topics.rst

```diff
@@ -203,3 +203,52 @@ composed using Findlib. Even though this is useful to run some (usually old)
 ppx's which don't support drivers, Jbuilder does not support preprocessing with
 ppx this way. but a workaround exists using the `ppxfind
 <https://github.com/diml/ppxfind>`_ tool.
+
+Linting
+=======
+
+jbuilder allows a user to defined a custom linting specification that acts like
+a preprocessor, but only needs to validate the sources the user has a written.
+The specification is defined by ``lint`` field. A ``lint`` field is allowed
+wherever a ``preprocess`` field is allowed (executable and library stanzas).
+This ``lint`` field follows the same `Preprocessing specification`_ that
+``preprocess`` accepts. The ``lint`` field will define the command that will run
+against the sources to make sure the sources are valid (by returning a 0 exit
+code), and will optionally write a .corrected file that will instruct the user
```

Comment:
BTW, at the moment this will not work very well until we finish the promote feature. Additionally I'm wondering if we shouldn't say indeed: either return a non-zero exit code or write a .corrected file. If jbuilder takes care of the diffing and promotion, there is no need for individual systems to perform the checks themselves and it's better if they don't do both, otherwise it's not clear in jbuilder whether we should report the non-zero exit code or report the diff.

On another note, ppx_driver currently names the corrected file `.ppx-corrected`. This is because several systems might generate correction and we can't have them all generate the same file. Maybe we should rename that to `.lint-corrected` so that it works with action preprocessors as well.

That makes me think that we shouldn't document this until it's all ready...

Reply:
Hmm, OK. Then I will merge the rest without the documentation commit. It's still nice to have this documentation somewhere to refer to. I've already found it useful to explain the feature to @avsm for example.